db/background_mgr.go

```diff
@@ -164,6 +347,40 @@ func (b *BackgroundManager) SetError(err error) {
 
 	b.lastError = err
 	b.State = BackgroundProcessStateError
+	b.Terminate()
+}
+
+// UpdateStatusClusterAware gets the current local status from the running process and updates the status document in
+// the bucket. Implements a retry. Used for Cluster Aware operations
+func (b *BackgroundManager) UpdateStatusClusterAware() error {
+	err, _ := base.RetryLoop("UpdateStatusClusterAware", func() (shouldRetry bool, err error, value interface{}) {
+		status, err := b.getStatusLocal()
+		if err != nil {
+			return true, err, nil
+		}
+
+		err = b.clusterAwareOptions.bucket.SetRaw(b.clusterAwareOptions.StatusDocID(), 0, status)
+		if err != nil {
+			return true, err, nil
+		}
+
+		return false, nil, nil
+	}, base.CreateSleeperFunc(5, 100))
+	return err
+}
+
+// UpdateHeartbeatDocClusterAware simply performs a touch operation on the heartbeat document to update its expiry.
+// Implements a retry. Used for Cluster Aware operations
+func (b *BackgroundManager) UpdateHeartbeatDocClusterAware() error {
+	err, _ := base.RetryLoop("", func() (shouldRetry bool, err error, value interface{}) {
```

Comment:
I don't think we'd want retry when touching the heartbeat - waiting for the next interval is usually sufficient.

Reply:
Okay although its fairly important to know if that doc is missing at any point as if that doc disappears for whatever reason it would allow another instance of the process to start up. Not sure how we can avoid that if we just wait for the next interval.